haas/model.py

```diff
@@ -172,7 +174,9 @@ def power_cycle(self):
             # just turn it on in that case. This way we can save power by
             # turning things off without breaking the HaaS.
             status = self._ipmitool(['chassis', 'power', 'on'])
-        return status == 0
+            if status != 0:
+                # If it still doesn't work, then it's a real error:
+                raise OBMError('Could not power cycle node %s' % node.label)
```

Comment:
Suggest::

    if 0 != self._ipmitool(['chassis', 'power', 'cycle']):
        return
    if 0 != self._ipmitool(['chassis', 'power', 'on']):
        return
    raise OBMError('Could not power cycle node %s' % node.label)